ci/unit/test_main_node.sh

```diff
@@ -22,9 +22,20 @@ sudo ln -sf "$SL_PREFIX/share/spdk/scripts/setup.sh" /usr/share/spdk/scripts
 sudo ln -sf "$SL_PREFIX/share/spdk/scripts/common.sh" /usr/share/spdk/scripts
 sudo ln -s "$SL_PREFIX/include"  /usr/share/spdk/include
 
-# set CMOCKA envs here
-export CMOCKA_MESSAGE_OUTPUT=xml
-export CMOCKA_XML_FILE="$DAOS_BASE"/test_results/%g.xml
 cd "$DAOS_BASE"
-IS_CI=true OLD_CI=false utils/run_test.sh
-./utils/node_local_test.py all
+if [ "$WITH_VALGRIND" = "memcheck" ]; then
+    echo "run_test_main_node.sh with memcheck"
+    # run_test.sh with valgrind memcheck
+    IS_CI=true OLD_CI=false RUN_TEST_VALGRIND=memcheck utils/run_test.sh
+    # Remove DAOS_BASE from memcheck xml results
+    sed -i "s:$DAOS_BASE::g" results-*-memcheck.xml
+    mkdir -p valgrind_memcheck_results
+    mv results-*-memcheck.xml valgrind_memcheck_results
```

Comment:
These three lines need to execute if the run_test.sh script fails.

Reply:
I just sent a patch to keep the execution even if run_test.sh fails, what do you think?
btw, Jenkins job #29 shows Valgrind results without the $DAOS_BASE in the process path, I think that's all the conversion required for the XML, right? (for the final patch I will put again the suppression file, so in Valgrind Results we should see nothing)

Reply:
I can't see anything in https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-3023/29/ that suggests the substituion is working.  https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-3023/29/valgrindResult/pid=3147,0x0/ shows something but it's from a system header file, none of our source.

That's not the only problem though, it looks like your change just runs "cmd |" which is not a syntax I'm aware of, I typically run "cmd || true" if the script is running -e and I want to ignore command result, but in this case if the command fails you want to run the xml modify part, then exit with the correct status, which would mean something like this

`
set +e
cmd
RC=$?
set -e
modify_xml
[ $RC -ne 0 ] && exit $RC
`

Alternatively, there may be a _post.sh script that you could move the modify_xml script to?

Reply:
> These three lines need to execute if the run_test.sh script fails.

If that's the case, then they should be moved to a `post { }` handler.

Reply:
Hey guys, sorry, I was confused before.

I checked this command in my terminal `RUN_TEST_VALGRIND=memcheck utils/run_test.sh ; echo $?` with the suppressions and with an empty supp file, and either way the retcode was zero and `results-*-memcheck.xml` were generated. The difference was that the first files don't have the leaks listed, and the latter have the leaks.

Instead of checking the retcode, for my next patch I'll verify the existence of `results-*-memcheck.xml` files. Let me know you're comments.

Reply:
> and either way the retcode was zero and `results-*-memcheck.xml` were generated

This is how tests are supposed to operate if they produce a results file.  The return code of the test reflects not whether the tests passed or failed but whether the test harness itself ran without any problems.  So if the test harness runs and completes without any issues, it should exit 0, even if some of the tests actually failed.